code/modules/overmap/overmap_object.dm

```diff
@@ -42,6 +58,132 @@
+		ny = low_edge
+	if((x == nx) && (y == ny))
+		return //we're not flying off anywhere
+
+	var/turf/T = locate(nx,ny,z)
+	if(T)
+		forceMove(T)
+
+/obj/effect/overmap/proc/is_still()
+	return !MOVING(speed[1], min_speed) && !MOVING(speed[2], min_speed)
+
+/obj/effect/overmap/proc/get_speed()
+	return round(sqrt(speed[1] ** 2 + speed[2] ** 2), SHIP_MOVE_RESOLUTION)
+
+/obj/effect/overmap/proc/get_heading()
+	var/res = 0
+	if(MOVING(speed[1], min_speed))
+		if(speed[1] > 0)
+			res |= EAST
+		else
+			res |= WEST
+	if(MOVING(speed[2], min_speed))
+		if(speed[2] > 0)
+			res |= NORTH
+		else
+			res |= SOUTH
+	return res
+
+/obj/effect/overmap/proc/adjust_speed(n_x, n_y)
+	CHANGE_SPEED_BY(speed[1], n_x, min_speed)
+	CHANGE_SPEED_BY(speed[2], n_y, min_speed)
+	update_icon()
+
+/obj/effect/overmap/proc/get_delta_v() //This is here for inheritance reasons.
+	return
+
+/obj/effect/overmap/proc/get_vessel_mass() //Same as above.
+	return vessel_mass
+
+/obj/effect/overmap/proc/can_burn()
+	if(halted)
+		return FALSE
+	if (world.time < last_burn + burn_delay)
+		return FALSE
+	else
+		return TRUE
+
+/obj/effect/overmap/Process()
+	if(!halted && !is_still() && can_move)
+		var/list/deltas = list(0,0)
+		for(var/i = 1 to 2)
+			if(MOVING(speed[i], min_speed))
+				position[i] += speed[i] * OVERMAP_SPEED_CONSTANT
+				if(position[i] < 0)
+					deltas[i] = ceil(position[i])
+				else if(position[i] > 0)
+					deltas[i] = Floor(position[i])
+				if(deltas[i] != 0)
+					position[i] -= deltas[i]
+					position[i] += (deltas[i] > 0) ? -1 : 1
+
+		update_icon()
+		var/turf/newloc = locate(x + deltas[1], y + deltas[2], z)
+		if(newloc && loc != newloc)
+			Move(newloc)
+			handle_wraparound()
+	handle_overmap_pixel_movement()
+
+/obj/effect/overmap/proc/accelerate(var/direction, var/accel_limit)
+	var/actual_accel_limit = accel_limit / KM_OVERMAP_RATE
+	if(can_burn())
+		last_burn = world.time
+		var/delta_v = get_delta_v() / KM_OVERMAP_RATE
+		var/partial_power = Clamp(actual_accel_limit / delta_v, 0, 1)
+		var/acceleration = min(get_delta_v(TRUE, partial_power) / KM_OVERMAP_RATE, actual_accel_limit)
+		if(direction & EAST)
+			adjust_speed(acceleration, 0)
+		if(direction & WEST)
+			adjust_speed(-acceleration, 0)
+		if(direction & NORTH)
+			adjust_speed(0, acceleration)
+		if(direction & SOUTH)
+			adjust_speed(0, -acceleration)
+
+
+/obj/effect/overmap/proc/decelerate()
+	if(((speed[1]) || (speed[2])) && can_burn())
+		if (speed[1])
+			var/partial_power = Clamp(speed[1] / (get_delta_v() / KM_OVERMAP_RATE), 0, 1)
+			var/delta_v = get_delta_v(TRUE, partial_power) / KM_OVERMAP_RATE
+			adjust_speed(-SIGN(speed[1]) * min(delta_v, abs(speed[1])), 0)
+		if (speed[2])
+			var/partial_power = Clamp(speed[2] / (get_delta_v() / KM_OVERMAP_RATE), 0, 1)
+			var/delta_v = get_delta_v(TRUE, partial_power) / KM_OVERMAP_RATE
+			adjust_speed(0, -SIGN(speed[2]) * min(delta_v, abs(speed[2])))
+		last_burn = world.time
+
+
+/obj/effect/overmap/proc/get_specific_wet_mass()
```

Comment:
this should get moved at least down to visitable, it's for wacky dynamic mass calculations, meaningless on objects without attached zlevels.